### PR TITLE
feat: Add compatibility with Quince release

### PR DESCRIPTION
**Pull request by bra-i-am**
## Depends on
According to the [eox-theming test cases document](https://docs.google.com/document/d/18urdJPitAu1lUBvYdQcGXm4WRDbKD2eDf4TTGgrlzZU/edit#heading=h.1n0md9yilzd7), to test this PR you have to clone **ednx-test-themes**; in this case because of some changes needed to keep compatibility with **Django 4** [you need to git checkout it to this PR's branch](https://github.com/eduNEXT/ednx-test-themes/pull/3)

## Description
This PR adds support to the Quince release updating the Django version, requirements, workflow dependencies, and fixing an issue with a test

### Take into account
This PR aims to solve the compatibility among eox-theming and Quince release; despite there are some issues with the `ednx-test-themes` templates, these are out of the scope of the [JIRA ISSUE](https://edunext.atlassian.net/browse/DS-777) and the solution will be aborded in future efforts

## Testing instructions

1. Install the plugin, you can: 
    - Use OPENEDX_EXTRA_PIP_REQUIREMENTS in the config.yml file
    - Add it with the `docker-compose.override.yml` to the container and execute `pip install <path/to/eox-theming>`.

2. Follow the instructions [here](https://docs.google.com/document/d/18urdJPitAu1lUBvYdQcGXm4WRDbKD2eDf4TTGgrlzZU/edit#).

## Additional information
[JIRA ISSUE DS-777](https://edunext.atlassian.net/browse/DS-777)

### CODE REVIEW

File: eox_theming/tests/theming/test_middleware.py [hunk 1]
```diff
@@ -20,6 +20,6 @@ def test_theme_selected(self, module_mock):
         module_mock.return_value.return_value = site_theme
 
         from eox_theming.theming.middleware import EoxThemeMiddleware  # pylint: disable=import-outside-toplevel
-        EoxThemeMiddleware().process_request(request)
+        EoxThemeMiddleware(get_response=lambda req: None).process_request(request)
```
Comment (bra-i-am) [inline eox_theming/tests/theming/test_middleware.py:23]: Used a placeholder because `get_response` was asked as mandatory during the test

**Review by dcoa — DISMISSED** [review]

**Comment by MaferMazu** [comment]
Thanks for this PR, @bra-i-am.

- [x] You update the chore requirements.
- [x] Updated the Github actions.
- [x] Update the requirements.
- [ ] Update the readme and other doc (The test cases are updated, but maybe we want to update the readme, too)
- [ ] The test cases work.

The test cases seem to work because when I tried, I obtained this:
Screenshot from 2024-01-23 18-12-39
But according to my understanding, the page should be purple, not red. I'm not sure why this is happening.

File: README.rst [hunk 2]
```diff
@@ -28,8 +28,13 @@ Compatibility Notes
...
 |      Palm        |       >= 6.0        |
 +------------------+---------------------+
+|      Quince      |       >= 7.0        |
++------------------+---------------------+
+
+**NOTES**: 
 
-**NOTE**: From Lilac version Django 2.2 is not supported, you should use Django 3.2 and eox-tenant >=4.0.
+- From Lilac version Django 2.2 is not supported, you should use Django 3.2 and eox-tenant >=4.0.
+- From Quince version Django 3.#.# is not supported, you should use Django 4.2
```
Comment (MaferMazu) [inline README.rst:37]: We still support Django 3.x; in our test we still have:
```yml
python-version: ["3.8", "3.10", "3.11"]
django: ["32","42"]
```
```suggestion
```

Comment (bra-i-am) [reply README.rst]: I'm so sorry about this, you are right! I already applied the correction

**Comment by bra-i-am** [comment]
hi @MaferMazu, the incorrect display of the page (red instead of purple) is directly related to issues in the templates of `endx-test-themes`. Thus, this is an out-of-scope issue that will be addressed in future efforts.

**Review by MaferMazu — APPROVED** [review]
Thanks, @bra-i-am.

**Pull request merged**